jina/serve/runtimes/request_handlers/data_request_handler.py

```diff
@@ -95,17 +96,24 @@ async def handle(self, requests: List['DataRequest']) -> DataRequest:
             ),
         )
         # assigning result back to request
-        if r_docs is not None:
-            if isinstance(r_docs, DocumentArray):
-                DataRequestHandler.replace_docs(requests[0], r_docs)
-            elif isinstance(r_docs, dict):
+        if return_data is not None:
+            if isinstance(return_data, DocumentArray):
+                DataRequestHandler.replace_docs(requests[0], return_data)
+            elif isinstance(return_data, dict):
                 params = requests[0].parameters
-                params.update(r_docs)
+                results_key = '__results__'
+
+                if not results_key in params.keys():
+                    params[results_key] = dict()
+
+                params[results_key].update({self.args.name: return_data})
```

Comment:
I think we also need `DataRequestHandler.replace_docs(requests[0], docs)` ? Otherwise one can not change docs in place when returning params.

Reply:
You are right something need to be done otherwise the docs are not replace as soon as the executor return a dict. Nevertheless it seems that this bug is not new, it is the failing on current master as well